### PR TITLE
Remove HLRC from security integration tests

### PR DESCRIPTION
**Pull request by slobodanadamovic**
This PR removes remaining usages of HLRC from security integration tests. 

Relates to https://github.com/elastic/elasticsearch/issues/83423

### CODE REVIEW

**Comment by slobodanadamovic** [comment]
@elasticsearchmachine run elasticsearch-ci/bwc
@elasticsearchmachine run elasticsearch-ci/part-1 
@elasticsearchmachine run elasticsearch-ci/part-1-fips
@elasticsearchmachine run elasticsearch-ci/part-3 
@elasticsearchmachine run elasticsearch-ci/part-3-fips

File: x-pack/qa/reindex-tests-with-security/src/yamlRestTest/java/org/elasticsearch/xpack/security/ReindexWithSecurityIT.java [hunk 1]
```diff
@@ -179,24 +188,43 @@ private void createIndicesWithRandomAliases(String... indices) throws IOExceptio
...
         }
-        refresh(restClient.getLowLevelClient(), String.join(",", indices));
+        request.setOptions(RequestOptions.DEFAULT);
+        return client().performRequest(request);
     }
 
-    private class TestRestHighLevelClient extends RestHighLevelClient {
-        TestRestHighLevelClient() {
-            super(client(), restClient -> {}, Collections.emptyList());
+    private static Response executeIgnoringConflicts(String method, String endpoint, String jsonBody, Map<String, String> parameters)
```
Comment (gwbrown) [inline x-pack/qa/reindex-tests-with-security/src/yamlRestTest/java/org/elasticsearch/xpack/security/ReindexWithSecurityIT.java:218]: Why is it necessary to ignore 409's in most of the places this is used, given we didn't do that before it looks like?

**Review by AthenaEryma — APPROVED** [review]
I left one question out of curiosity but LGTM!

File: x-pack/qa/reindex-tests-with-security/src/yamlRestTest/java/org/elasticsearch/xpack/security/ReindexWithSecurityIT.java (hunk 1 above)
Comment (slobodanadamovic) [reply x-pack/qa/reindex-tests-with-security/src/yamlRestTest/java/org/elasticsearch/xpack/security/ReindexWithSecurityIT.java:218]: These tests are failing if 409 status is not ignored. I simply replicated the behaviour of RestHighLevelClient. Both [updateByQuery](https://github.com/elastic/elasticsearch/blob/0799663a589aec510877272d41044f276e22fd2d/client/rest-high-level/src/main/java/org/elasticsearch/client/RestHighLevelClient.java#L388) and [deleteByQuery](https://github.com/elastic/elasticsearch/blob/0799663a589aec510877272d41044f276e22fd2d/client/rest-high-level/src/main/java/org/elasticsearch/client/RestHighLevelClient.java#L406) ignore 409 status code and simply return a response without throwing an exception. 

https://github.com/elastic/elasticsearch/blob/0799663a589aec510877272d41044f276e22fd2d/client/rest-high-level/src/main/java/org/elasticsearch/client/RestHighLevelClient.java#L394

**Pull request merged**